code/crates/core-consensus/src/handle/sync.rs

```diff
@@ -12,28 +12,40 @@ pub async fn on_value_response<Ctx>(
 where
     Ctx: Context,
 {
-    debug!(
-        value.certificate.height = %value.certificate.height,
+    let consensus_height = state.driver.height();
+    let cert_height = value.certificate.height;
+
+    if consensus_height < cert_height {
+        debug!(%consensus_height, %cert_height, "Received value response for higher height, queuing for later");
+
+        state.buffer_input(cert_height, Input::SyncValueResponse(value));
+
+        return Ok(());
+    }
+
+    info!(
+        value.certificate.height = %cert_height,
         signatures = value.certificate.commit_signatures.len(),
         "Processing value response"
     );
 
-    if state.driver.height() < value.certificate.height {
-        debug!("Received value response for higher height, queuing for later");
+    let proposer = state
+        .get_proposer(cert_height, value.certificate.round)
```

Comment:
curious if we perform any additional checks on `cert_height`. like if a peer sends a certificate with `height=0`, what will happen?

Reply:
oh, great catch @melekes...lost the check with my changes. Let me kook again, I think there's another one I missed.